client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -4,23 +4,82 @@ import { deleteTea, getTeas } from "../../../actions/teaActions";
 import TeaList from "./TeaList";
 
 export class TeaListContainer extends React.Component {
+  state = {
+    sortedIDs: [],
+    memoizedIDs: {},
+    sortColumn: "",
+    sortOrder: ""
+  };
+
   handleDeleteClick = tea => {
     this.props.handleDelete(tea);
   };
 
+  handleSortClick = (columnName, sortOrder) => {
+    const list = this.props.teas.teaIDs;
+    let currentState = this.state.memoizedIDs;
+    let newSortOrder;
+
+    if (this.state.memoizedIDs[columnName]) {
+      newSortOrder = this.state.memoizedIDs[columnName][sortOrder];
+    } else {
+      let mapped = list.map((el, i) => {
+        return {
+          index: i,
+          value: this.props.teas.allTeas[el][columnName]
+        };
+      });
+
+      mapped.sort((a, b) => {
+        if (a.value > b.value) {
+          return 1;
+        }
+        if (a.value < b.value) {
+          return -1;
+        }
+        return 0;
+      });
+
+      newSortOrder = mapped.map(el => list[el.index]);
+      const revSortOrder = [...newSortOrder].reverse();
+
+      currentState[columnName] = { asc: newSortOrder, desc: revSortOrder };
```

Comment:
Were you planning on allowing for multiple column sorting (e.g. primary, secondary, etc.), or just one column at a time? If you do decide to implement multiple sorting, this sort of memoization will get more complicated.

Reply:
Single column sorting will be sufficient for this project. Once filtering is added, that will enable users to narrow things down further and sort from there with enough granularity for this purpose.